src/lamp_py/migrations/versions/performance_manager_prod/002_f09e853d5672_update_prod_stop_sync.py

```diff
@@ -0,0 +1,418 @@
+                        static_canon.static_version_key AS static_version_key,
+                        ((static_canon.stop_sequence - zero_seq_vals.seq_adjust) - sync_adjust_vals.min_sync) + sync_adjust_vals.min_seq AS sync_stop_sequence
+                    FROM
+                        (
+                        SELECT
+                            canon_trips.direction_id AS direction_id,
+                            canon_trips.trunk_route_id AS trunk_route_id,
+                            canon_trips.route_id AS route_id,
+                            static_stops.parent_station AS parent_station,
+                            ROW_NUMBER() OVER (PARTITION BY canon_trips.static_version_key,
+                            canon_trips.direction_id,
+                            canon_trips.route_id
+                        ORDER BY
+                            static_stop_times.stop_sequence) AS stop_sequence,
+                            canon_trips.static_version_key AS static_version_key
+                        FROM
+                            (
+                            SELECT
+                                DISTINCT ON
+                                (COALESCE(static_trips.branch_route_id,
+                                static_trips.trunk_route_id),
+                                static_route_patterns.direction_id,
+                                static_route_patterns.static_version_key) static_route_patterns.direction_id AS direction_id,
+                                static_route_patterns.representative_trip_id AS representative_trip_id,
+                                static_trips.trunk_route_id AS trunk_route_id,
+                                COALESCE(static_trips.branch_route_id,
+                                static_trips.trunk_route_id) AS route_id,
+                                static_route_patterns.static_version_key AS static_version_key
+                            FROM
+                                static_route_patterns
+                            JOIN static_trips ON
+                                static_route_patterns.representative_trip_id = static_trips.trip_id
+                                AND static_route_patterns.static_version_key = static_trips.static_version_key
+                            WHERE
+                                static_route_patterns.static_version_key = {static_version_key}
+                                AND (static_route_patterns.route_pattern_typicality = 1
+                                    OR static_route_patterns.route_pattern_typicality = 5)
+                            ORDER BY
+                                COALESCE(static_trips.branch_route_id,
+                                static_trips.trunk_route_id),
+                                static_route_patterns.direction_id,
+                                static_route_patterns.static_version_key,
+                                static_route_patterns.route_pattern_typicality DESC) AS canon_trips
```

Comment:
is there a way that this could be a CTE instead of being repeated throughout this query?

not a big deal, just trying to think about how these queries work with best practices.

Reply:
Unfortunately, CTE's have a performance penalty in this application, since all results of a CTE have to be materialized before a query can be executed. In-lining the entire query allows the SQL engine to optimize the query, which is supposed to improve query performance. 

Risking the performance penalty of CTE's for this could significantly extend the migration duration.